### PR TITLE
Adding Igbo NER data 

### PR DESCRIPTION
**Pull request by purvimisal**
This PR adds the Igbo NER dataset.
Data: https://github.com/IgnatiusEzeani/IGBONLP/tree/master/ig_ner

### CODE REVIEW

**Comment by rkc007** [comment]
Thanks for the PR @purvimisal. 

Few comments below.

File: datasets/igbo_ner/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+
+## Dataset Description
+
+- **Homepage:**
+- **Repository:**
+- **Paper:**
```
Comment (rkc007) [inline datasets/igbo_ner/README.md:51]: Can you add `Homepage`, `Repository` and `Paper` here ?

File: datasets/igbo_ner/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+### Dataset Curators
+
+[More Information Needed]
+
+### Licensing Information
+
+[More Information Needed]
+
+### Citation Information
+
```
Comment (rkc007) [inline datasets/igbo_ner/README.md:136]: Can you add Citation information here ?

Comment (purvimisal) [reply datasets/igbo_ner/README.md:134]: Yeah added it. PTAL!

File: datasets/igbo_ner/README.md (hunk 1 above)
Comment (purvimisal) [reply datasets/igbo_ner/README.md]: Sure, added it.

**Comment by purvimisal** [comment]
Hi @rkc007, made the changes. PTAL!

File: datasets/igbo_ner/igbo_ner.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        dictionary = {}
+        with open(filepath, "r", encoding="utf-8-sig") as f:
+            if self.config.name == "ner_data":
+                for id_, row in enumerate(f):
+                    row = row.split("\t")
```
Comment (lhoestq) [inline datasets/igbo_ner/igbo_ner.py:106]:
```suggestion
                    row = row.strip().split("\t")
```
remove trailing `\n`

File: datasets/igbo_ner/igbo_ner.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                        dictionary[content_n]["sentences"] = []
+                    yield id_, {
+                        "content_n": content_n,
+                        "named_entity": dictionary[content_n]["named_entity"],
+                        "sentences": dictionary[content_n]["sentences"],
+                    }
+            else:
+                for id_, row in enumerate(f):
+                    yield id_, {
+                        "sentences": row,
```
Comment (lhoestq) [inline datasets/igbo_ner/igbo_ner.py:122]:
```suggestion
                        "sentences": row.strip(),
```

File: datasets/igbo_ner/README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
+
+### Data Fields
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/igbo_ner/README.md:73]: can you give a brief description of the data fields here please ? (content_n, named_entity, sentences)

File: datasets/igbo_ner/README.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+### Languages
+
+[More Information Needed]
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/igbo_ner/README.md:69]: Can you add one example from the dataset here please ?

You can get one by using `load_dataset` and taking the first sample `dataset["train"][0]` for example

**Review by lhoestq — COMMENTED** [review]
Nice thank !

I left a few comments, mostly to fill in a little bit more the dataset card

**Comment by purvimisal** [comment]
Hi, @lhoestq Thank you for the review. I have made all the changes. PTAL!

File: datasets/igbo_ner/README.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+### Languages
+
+[More Information Needed]
+
+## Dataset Structure
+
+### Data Instances
+
+{'content_n': 'content_0', 'named_entity': 'Ike Ekweremmadụ', 'sentences': ['Ike Ekweremmadụ', "Ike ịda jụụ otụ nkeji banyere oke ogbugbu na-eme n'ala Naijiria agwụla Ekweremmadụ"]}
```
Comment (lhoestq) [inline datasets/igbo_ner/README.md:69]:
````suggestion
Here is an example from the dataset:
```
{'content_n': 'content_0', 'named_entity': 'Ike Ekweremmadụ', 'sentences': ['Ike Ekweremmadụ', "Ike ịda jụụ otụ nkeji banyere oke ogbugbu na-eme n'ala Naijiria agwụla Ekweremmadụ"]}
```
````

adding code block

File: datasets/igbo_ner/README.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+### Data Fields
+
+content_n : ID 
+named_entity : Name of the entity 
+sentences : List of sentences for the entity
```
Comment (lhoestq) [inline datasets/igbo_ner/README.md:75]:
```suggestion
- content_n : ID 
- named_entity : Name of the entity 
- sentences : List of sentences for the entity 
```
adding bullet points

**Review by lhoestq — APPROVED** [review]
Nice thanks :)

**Comment by lhoestq** [comment]
the CI error is not related to your dataset, merging

**Pull request merged**